discovery/spec.md

```diff
@@ -974,31 +916,71 @@ Content-Type: application/json
 }
 ```
 
-#### `DELETE /services/{id}`
+#### `DELETE /services/{id}[?epoch={epoch}]`
 
 This MUST delete the Service at the referenced URL.
 
-The request MAY include a body. Any such body MUST be an object containing an
-`id` matching the `{id}` in the path. That object MAY include an `epoch`
-value. If the Service corresponding with the specified `id` has a higher
-recorded `epoch` value than given, the request MUST fail.
-
-If other service attributes are included in the request, those MUST be
-ignored for the purposes of processing the request.
+The request URL MAY include an OPTIONAL `epoch` query parameter, and if
+present, the Discovery Endpoint MUST reject the request if the corresponding
+Service has a different `epoch` value than the query parameter.
 
 The follow responses are defined by this specification:
-- `200 OK` if the new Service was deleted.
-  - The HTTP Response Body MUST include the Service values at the time of
-    deletion.
-- `400 Bad Request` if `id` is in the body and does not match the `{id}` in the
-  path
-- `404 Not Found` if there is no Service with the specified `id`
-- `409 Conflict` if the given epoch was less than or equal to the existing
-  epoch of the Service with the given `id`
+- `200 OK` if the Service was deleted or does not exist.
+  - The HTTP Response Body MUST include a Service definition with at least
+    its `id` attribute. It SHOULD include the remaining Service attributes
+	as they existed at the time of processing the request if possible.
+	Note that if the Service was deleted prior to processing of this request
+	the Service values might not be available to be returned.
+- `409 Conflict` if the given `epoch` was not equal to the existing `epoch` of
+   the Service.
 
 Other responses are allowed, but not defined by this specification.
 
-The format of the OPTIONAL HTTP Request body MUST adhere to the following:
+The format of the `200 OK` HTTP Response MUST adhere to the following:
+```
+200 OK
+Content-Type: application/json
+
+{
+  "url": "{url}",
+  "name": "{name}",
+  ... remainder of Service attributes ...
+}
+```
+
+#### `DELETE /services`
```

Comment:
Usually interactions with the collection are grouped together.  I would prefer this move this back up between POST and PUT.